### PR TITLE
Fixed #5368: enhanced keybinding search to support different key orders

### LINKED ISSUE #5368: [settings] Keybinding search should ignore the key order
In `Keyboard Shortcuts` UI, when we do a keybinding search, the results should include all combinations of keys. For example, searching `alt + shift` should be equivalent to searching `shift + alt`

In Theia:
Screenshot from 2019-06-05 16-15-53
Screenshot from 2019-06-05 16-21-22

In VS Code:
Screenshot from 2019-06-05 16-16-04

### PR DESCRIPTION
**Pull request by fangnx**
Fixed #5368.

Updated the `doSearchKeybindings` function to support matching identical keybinding patterns with different orders (e.g., `alt+shift` & `shift+alt`). Fuzzy search in keybinding patterns is still supported.

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
@fangnx do we now support the highlighting for keys that are out of order?
I tried, and I don't see the highlighting for such keys ( ex: <kbd>shift</kbd>+<kbd>alt</kbd> when searching <kbd>alt</kbd>+<kbd>shift</kbd> )

<div align='center'>

image


</div>

File: packages/keymaps/src/browser/keybindings-widget.tsx [hunk 1]
```diff
@@ -93,6 +93,17 @@ export class KeybindingWidget extends ReactWidget {
         this.focusInputField();
     }
 
+    // Calculate the net rendered result
```
Comment (vince-fugnitto) [inline packages/keymaps/src/browser/keybindings-widget.tsx]: Can we update the documentation to be consistent with out [coding guidelines](https://github.com/theia-ide/theia/wiki/Coding-Guidelines#comments).
We use [JSDoc](https://en.wikipedia.org/wiki/JSDoc) style documentation which you can see an example of here:

https://github.com/theia-ide/theia/blob/798ecc9304f395103e804d705a86565c14507364/packages/cpp/src/browser/cpp-build-configurations.ts#L74-L79

**Comment by vince-fugnitto** [comment]
@elaihau do you mind testing the PR as well?

**Comment by vince-fugnitto** [comment]
> See the part highlighted by yellow (yellow was added by me)

Yes we noticed that error too, it has to do with key chords which were originally not supported well in Theia. The error means there is no spacing between the chords, but we should fix it as part of a separate issue :)

**Comment by vince-fugnitto** [comment]
> works well. Thank you for the PR !
> 
> i found a UI bug while doing testing. not related to this PR though.
> 
> keybinding
> 
> See the part highlighted by yellow (yellow was added by me)

@fangnx are you okay opening up an issue for this problem on master and addressing it after this PR is merged?

**Comment by fangnx** [comment]
> > works well. Thank you for the PR !
> > i found a UI bug while doing testing. not related to this PR though.
> > keybinding
> > See the part highlighted by yellow (yellow was added by me)
> 
> @fangnx are you okay opening up an issue for this problem on master and addressing it after this PR is merged?

Sure ;)

**Pull request merged**